### PR TITLE
feat(builtin-audit-trails): added built-in section for audit trail

### PR DESCRIPTION
**Pull request by truszkowski**
# Description of the change

Added built-in audit trails feature. 
Now, we can view audit trail logs in Spacelift. 

## Checklist

Please make sure that the proposed change checks all the boxes below before requesting a review:

- [ ] I have reviewed the [guidelines for contributing](https://github.com/spacelift-io/user-documentation/blob/main/CONTRIBUTING.md) to this repository.
- [ ] The preview looks fine.
- [ ] The tests pass.
- [ ] The commit history is clean and meaningful.
- [ ] The pull request is opened against the `main` branch.
- [ ] The pull request is no longer marked as a draft.
- [ ] You agree to license your contribution under the [MIT license](https://github.com/spacelift-io/user-documentation/blob/main/LICENSE) to Spacelift (not required for Spacelift employees).
- You have updated the navigation files correctly:
    - [ ] No new pages have been added, or;
    - [ ] Only _nav.yaml_ has been updated because the changes only apply to SaaS, or;
    - [ ] Only _nav.self-hosted.yaml_ has been updated because the changes only apply to Self-Hosted, or;
    - [ ] Both _nav.yaml_ and _nav.self-hosted.yaml_ have been updated.

If the proposed change is ready to be merged, please request a review from `@spacelift-io/solutions-engineering`. Someone will review and merge the pull request.

_Spacelift employees should request reviews from the relevant engineers and are allowed to merge pull requests after they got at least one approval._

Thank you for your contribution! 🙇

### CODE REVIEW

**Review by adamconnelly — COMMENTED** [review]
I wonder if it would make sense to move the audit trail page into the "Main Concepts" section now that it's not just the webhook integration.

@jmfontaine @Daniellem97 what do you think?

**Comment by jmfontaine** [comment]
> I wonder if it would make sense to move the audit trail page into the "Main Concepts" section now that it's not just the webhook integration.

I don't think it suits the Main Concepts section. It is not really a concept and it is not central to understanding Spacelift.

I would keep it where it is since it still is an integration but with added retention.

File: docs/integrations/audit-trail.md [hunk 1]
```diff
@@ -5,9 +5,32 @@
 
 Spacelift optionally supports auditing all operations that change Spacelift resources. This is handled by asynchronously sending webhooks to a user-supplied endpoint.
 
+## Built-in logs
+
+As an admin, you can view Audit trail logs by navigating to the Audit trail section of your account settings and choosing the _logs_ tab:
+
+![](<../assets/screenshots/audit-trail-logs.png>)
+
+Now, you can look for specific events using filters (on the left side) and date picker (in the top-right corner).
```
Comment (adamconnelly) [inline docs/integrations/audit-trail.md:14]:
```suggestion
You can look for specific events using filters (on the left side) and the date picker (in the top-right corner).
```

File: docs/integrations/audit-trail.md [hunk 2]
```diff
@@ -5,9 +5,32 @@
...
+
+![](<../assets/screenshots/audit-trail-logs-filters.png>)
+
+You can see the details:
+
+![](<../assets/screenshots/audit-trail-logs-details-button.png>)
+![](<../assets/screenshots/audit-trail-logs-details.png>)
+
+Also, go to the affected resource or apply another filter:
```
Comment (adamconnelly) [inline docs/integrations/audit-trail.md:23]:
```suggestion
You can also go to the affected resource or apply another filter:
```

File: docs/integrations/audit-trail.md [hunk 3]
```diff
@@ -5,9 +5,32 @@
...
+
+![](<../assets/screenshots/audit-trail-logs-dropdown.png>)
+
+### Retention
+
+Logs are kept for 30 days.
+
 ## Setup
```
Comment (adamconnelly) [inline docs/integrations/audit-trail.md:31]: Maybe we should rename this to "Webhooks" now?

File: docs/integrations/audit-trail.md [hunk 4]
```diff
@@ -5,9 +5,32 @@
 
 Spacelift optionally supports auditing all operations that change Spacelift resources. This is handled by asynchronously sending webhooks to a user-supplied endpoint.
```
Comment (adamconnelly) [inline docs/integrations/audit-trail.md:6]: I think we need to change this sentence, because we no-longer just send a webhook - we also have the built-in logs. How about:

> Spacelift supports auditing all operations that change Spacelift resources. We provide a built-in audit log as well as webhook functionality to allow you to optionally store your audit logs in a third party system.

**Comment by adamconnelly** [comment]
> > I wonder if it would make sense to move the audit trail page into the "Main Concepts" section now that it's not just the webhook integration.
> 
> I don't think it suits the Main Concepts section. It is not really a concept and it is not central to understanding Spacelift.
> 
> I would keep it where it is since it still is an integration but with added retention.

@jmfontaine that's a good point about it not being a concept. The reason I suggested moving it though is that it's no-longer just an integration since we're providing audit functionality built-into the product, along with optionally being able to integrate with an external system. So the default is that you will get auditing without having to configure any integration.

Maybe the "Product" section would make more sense actually? But fair enough if you'd rather not move it - it's in a bit of a grey zone since it's half integration, half not.

**Comment by jmfontaine** [comment]
> So the default is that you will get auditing without having to configure any integration.
@adamconnelly For some plans, not for all, so the integration is still the way to go for some users.

I hear you about this feature being in a grey area but I cannot think of a better location for this page at this time. 😅

**Comment by jmfontaine** [comment]
@adamconnelly I agree that this is a grey area, but I cannot think of a better location for this information.

**Review by Daniellem97 — DISMISSED** [review]

File: docs/integrations/audit-trail.md [hunk 5]
```diff
@@ -5,11 +5,34 @@
...
+
+![](<../assets/screenshots/audit-trail-logs-dropdown.png>)
+
+### Retention
+
+Logs are kept for 30 days.
+
+## Webhook Setup
+
+In order to set up the audit trail, navigate to the Audit trail section of your account settings and choosing the _configuration_ tab, and click the _Set up_ button:
```
Comment (peterdeme) [inline docs/integrations/audit-trail.md:35]: This was the old wording, I'd rephrase it a bit:

```suggestion
In order to set up the webhooks for audit events, navigate to the Audit trail section of your account settings and choosing the _configuration_ tab, and click the _Set up_ button:
```

**Review by peterdeme — APPROVED** [review]
one nit

**Review by adamconnelly — DISMISSED** [review]
I agree with Peter's rewording suggestion about the webhooks part, but other than that LGTM.

File: docs/integrations/audit-trail.md (hunk 5 above)
Comment (truszkowski) [reply docs/integrations/audit-trail.md]: changed 🙇‍♂️
